engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -1810,7 +1810,7 @@ namespace dmGameSystem
                         RenderParticlefxNodes(scene, entries + start, node_transforms + start, node_opacities + start, stencil_scopes + start, n, gui_context);
                         break;
                     case dmGui::NODE_TYPE_CUSTOM:
-                        RenderCustomNodes(scene, custom_type, GetCompGuiCustomType(gui_world->m_CompGuiContext, custom_type), entries + start, node_transforms + start, node_opacities + start, stencil_scopes + start, n, gui_context);
+                        RenderCustomNodes(scene, prev_custom_type, GetCompGuiCustomType(gui_world->m_CompGuiContext, prev_custom_type), entries + start, node_transforms + start, node_opacities + start, stencil_scopes + start, n, gui_context);
```

Comment:
Caused an assert previously.
Since it's a "flush" operation, triggered by the change in render type, we need to use the _previously_ collected type.